articles/140_topic_and_service_name_mapping.md

```diff
@@ -243,41 +221,72 @@ Here is a non-exhaustive list of prefixes:
 | ROS Parameter        | rp     |
 | ROS Action           | ra     |
 
-While all planned prefixes consist of two characters, i.e. `rX`, anything proceeding the first namespace separator, i.e. `__`, can be considered part of the prefix.
+While all planned prefixes consist of two characters, i.e. `rX`, anything proceeding the first namespace separator, i.e. `/`, can be considered part of the prefix.
 The standard reserves the right to use up to 8 characters for the prefix in case additional prefix space is needed in the future.
 
-### Communicating with Non-ROS Topics
+#### Communicating with Non-ROS Topics
 
 Since all ROS topics are prefixed when being converted to DDS topic names, it makes it impossible to subscribe to existing DDS topics which do not follow the same naming pattern.
 For example, if an existing DDS program is publishing on the `image` topic (and is using the DDS equivalent to the ROS message type) then a ROS program could not subscribe to it because of the name mangling.
 Therefore to allow ROS programs to interoperate with "native" DDS topic names the API should provide a way to skip the ROS specific prefixing.
 
-### ROS to DDS Name Conversion Examples
+### DDS Partitions as namespace hierarchy
```

Comment:
Throughout this you use "namespace hierarchy", this might be a of a nitpick but I see this as a redundant term. From my experience, multi-element namespaces are hierarchical by design. So to me it would be clearer to just say "namespace" instead. Also, the notion of splitting namespaces into parts with the `/` is already covered in sections above, so the hierarchy within the namespace is an established concept already.

For this title I'd reword it to something like "ROS Namespaces as DDS Partitions" (think of it as, "we already have ROS namespaces which are described above, now we'll explain how we'll implement them with DDS partitions"). Also, I think this section should go before the "ROS Specific Name Prefix" subsection and the "Communicating with Non-ROS Topics" subsection, basically it should be the first subsection of the "Mapping of ROS 2 Topic and Service Names to DDS Topics" section. Because I think it is probably confusing to talk about the ROS prefix and other special cases before explaining that we will use DDS partitions and what the partitions are. (so also move the "DDS Partitions Characteristics" subsection up there as well)